### PR TITLE
refactor(es/minifier): Respect top-level when invoking IIFE 

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

Follow up of #7671 

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc/tests/fixture/issues-2xxx/2679/output/index.js [hunk 1]
```diff
@@ -1,4 +1,2 @@
-!function() {
-    var a = {};
-    a.b = 1, a = null, window.test = a;
-}();
+var a;
```
Comment (jridgewell) [inline crates/swc/tests/fixture/issues-2xxx/2679/output/index.js:1]: This is leaking `a` into the global variable list. Is that ok?

**Review by jridgewell — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_minifier

File: crates/swc/tests/fixture/issues-2xxx/2679/output/index.js (hunk 1 above)
Comment (Austaras) [reply crates/swc/tests/fixture/issues-2xxx/2679/output/index.js:1]: Yes, it's in es module.

**Review by kdy1 — COMMENTED** [review]
reset ci

**Pull request merged**